Tasks/DockerV2/dockerbuild.ts

```diff
@@ -73,3 +90,133 @@ export function run(connection: ContainerConnection, outputUpdate: (data: string
         outputUpdate(taskOutputPath);
     });
 }
+
+function GetImageAnnotation(connection: ContainerConnection, dockerFile: string ): ImageAnnotations {
+    let imageAnnotations: ImageAnnotations = {
+        BaseImageName :"",
+        BaseImageDigest: ""
+    };
+
+    imageAnnotations.BaseImageName = getBaseImageName(dockerFile);
+
+    if (imageAnnotations.BaseImageName && imageAnnotations.BaseImageName != ""){
+         imageAnnotations.BaseImageDigest = getImageDigest(connection, imageAnnotations.BaseImageName);
+    }
+
+    return imageAnnotations;
+}
+
+function getBaseImageName(dockerFilePath: string): string {
```

Comment:
Some of this functions are going to be reused in further PRs to add support for V0 and V1. Is there an easy way where I can this functions that is shared between the docker tasks?

Reply:
I think one way to write this can be:
- Modify [getDefaultLabels](https://github.com/microsoft/azure-pipelines-tasks/blob/master/common-npm-packages/docker-common-v2/pipelineutils.ts#L70) signature to:
`getDefaultLabels(addPipelineData?: boolean, addBaseImageData?: boolean)`
- Add a methods `getBaseImageName` and `getImageDigest` in [containerimageutils](https://github.com/microsoft/azure-pipelines-tasks/blob/master/common-npm-packages/docker-common-v2/containerimageutils.ts)
- Add method `addBaseImageLabels` in `pipelineutils` which calls `getBaseImageName` and `getImageDigest`

Thoughts?

Reply:
Couple of questions:
- If getDefaultLabels is the responsible to add the labels then, is going to need the dockerFile path as well.
- The containerimageutils already has a method call getBaseImageName but I cannot used because it doesn't work well with multi-stage builds. I could replace that method with the one from this PR and the result should be the same.
- Is there no problems using the pipelineutils in DockerV0 and DockerV1 they both needs the getBaseImageName and getImageDigest as well.

Reply:
My thoughts:

> If getDefaultLabels is the responsible to add the labels then, is going to need the dockerFile path as well.

We can add DockerfilPath as the third argument:
`getDefaultLabels(addPipelineData?: boolean, addBaseImageData?: boolean, DockerfilePath?: string)`

> The containerimageutils already has a method call getBaseImageName but I cannot used because it doesn't work well with multi-stage builds. I could replace that method with the one from this PR and the result should be the same.

I have reservations against changing the existing behaviour. Is there any official definition of what should be considered as the base image in a multi-stage build?
I am thinking, adding `getBaseImageName2` might be another option?
@ds-ms thoughts on this?

> Is there no problems using the pipelineutils in DockerV0 and DockerV1 they both needs the getBaseImageName and getImageDigest as well.

It should not be an issue. They can both take dependency on `pipelineutils`

Reply:
> I am thinking, adding getBaseImageName2 might be another option?

What do you think `getFinalBaseImage` ?

Reply:
Let continue this conversation in this PR https://github.com/microsoft/azure-pipelines-tasks/pull/14803